syne_tune/backend/sagemaker_backend/sagemaker_backend.py

```diff
@@ -522,8 +529,12 @@ def delete_checkpoint(self, trial_id: int):
     def set_path(
         self, results_root: Optional[str] = None, tuner_name: Optional[str] = None
     ):
-        # we use the tuner-name to set the checkpoint directory
-        self.tuner_name = tuner_name
+        """
+        For this backend, it is mandatory to call this method passing ``tuner_name``
+        before the backend is used. ``results_root`` is ignored here.
+        """
+        if tuner_name is not None:
```

Comment:
This is the bugfix. This is called in `remote_main.py`, and it set `self.tuner_name` back to `None`, which is clearly wrong.